aws.tf

```diff
@@ -27,7 +27,7 @@ EOF
 }
 
 resource "aws_iam_role_policy_attachment" "cluster_autoscaler" {
-  count = var.aws_create_iam_policy && var.aws_iam_role_for_policy != null ? 1 : 0
+  count = var.aws_create_iam_policy ? 1 : 0
```

Comment:
Then make aws_iam_role_for_policy mandatory.
And separate it to the second commit because it is BC.

Reply:
I was getting strange behaviour without this change, however my fix here is obviously wrong. I'm going to update this pull request without this change in it, and when I next do a full build I'll work out what I was actually trying to fix.